src/dbnode/persist/fs/retriever.go

```diff
@@ -309,17 +312,65 @@ func (r *blockRetriever) processIndexChecksumRequest(
 	req.onIndexChecksumCompleted(entry)
 }
 
+func (r *blockRetriever) processReadMismatchRequest(
+	req *retrieveRequest,
+	seeker ConcurrentDataFileSetSeeker,
+	seekerResources ReusableSeekerResources,
+) {
+	mismatch, err := seeker.SeekReadMismatchesByIndexChecksum(
+		req.indexChecksum, req.mismatchChecker, seekerResources)
+
+	if err != nil && err != errSeekIDNotFound {
+		req.onError(err)
+		return
+	}
+
+	if err == errSeekIDNotFound {
+		req.onIndexMismatchCompleted(wide.ReadMismatch{})
+		return
+	}
+
+	req.onIndexMismatchCompleted(mismatch)
+	req.onCallerOrRetrieverDone()
```

Comment:
This looks correct, but I think `processIndexChecksumRequest` is missing this? 

It should also call `req.onCallOrRetrieverDone()` after `req.onIndexChecksumCompleted(entry)` yeah?

Reply:
Good catch, made a note that I had to add it there but then completely forgot to actually do it